app/src/runtime_arg.c

```diff
@@ -0,0 +1,152 @@
+    uint8_t dummy_type = 0;
+    uint8_t dummy_internal = 0;
+    for (uint32_t i = 0; i < deploy_argLen; i++) {
+        //key
+        CHECK_PARSER_ERR(parse_item(ctx));
+
+        //value
+        CHECK_PARSER_ERR(parse_item(ctx));
+        //type
+        CHECK_PARSER_ERR(get_type(ctx, &dummy_type, &dummy_internal));
+
+    }
+    return parser_ok;
+}
+
+
+parser_error_t searchRuntimeArgs(char *argstr, uint8_t *type, uint8_t *internal_type, uint32_t deploy_argLen, parser_context_t *ctx) {
+    uint16_t start = ctx->offset;
+    char buffer[300];
+    MEMZERO(buffer, 300);
+    uint8_t dummy_type = 0;
+    uint8_t dummy_internal = 0;
+
+    for (uint32_t i = 0; i < deploy_argLen; ++i) {
+        //key
+        CHECK_PARSER_ERR(copy_item_into_charbuffer(ctx, buffer, sizeof(buffer)));
+        if (strcmp(buffer, argstr) == 0) {
+              //value
+            CHECK_PARSER_ERR(parse_item(ctx));
+
+            CHECK_PARSER_ERR(get_type(ctx, type, internal_type));
+
+            ctx->offset = start;
+            return parser_ok;
+        }
+        //value
+        CHECK_PARSER_ERR(parse_item(ctx));
+
+        CHECK_PARSER_ERR(get_type(ctx, &dummy_type, &dummy_internal));
+    }
+
+    // runtimarg_notfound is an expected error, so we should
+    // set the offset to its original value back for further processing
+    ctx->offset = start;
+    return parser_runtimearg_notfound;
+}
+
+parser_error_t showRuntimeArgsHash(__Z_UNUSED ExecutableDeployItem item, parser_context_t *ctx,
+                                          uint32_t bytes_len, char *name, uint8_t name_len,
+                                          char *outKey, uint16_t outKeyLen,
+                                          char *outVal, uint16_t outValLen,
+                                          uint8_t pageIdx, uint8_t *pageCount) {
+
+    uint8_t hash[BLAKE2B_256_SIZE];
+    MEMZERO(hash, BLAKE2B_256_SIZE);
+
+    if (blake2b_hash(ctx->buffer + ctx->offset, bytes_len, hash) != zxerr_ok){
+        return parser_unexepected_error;
+    };
+
+    snprintf(outKey, outKeyLen, "Args hash");
+
+    // name-hash
+    // name + '-' + hex-hash + 'null-terminator'
+    uint32_t output_len = name_len + 1 + (BLAKE2B_256_SIZE * 2) + 1;
+    uint8_t output[output_len];
+    MEMZERO(output, output_len);
+
+    char hex_hash[BLAKE2B_256_SIZE * 2];
+    encode_hex((char *)hash, BLAKE2B_256_SIZE, hex_hash);
+
+    MEMCPY(output, name, name_len);
+    output[name_len] = '-';
+    MEMCPY((output + name_len + 1), hex_hash, BLAKE2B_256_SIZE * 2);
+
+    pageString(outVal, outValLen, (char *)output, pageIdx, pageCount);
+
+    return parser_ok;
+}
+
+parser_error_t showRuntimeArgByIndex(uint16_t index, char *outKey, uint16_t outKeyLen, char *outVal, uint16_t outValLen,
+        uint16_t pageIdx, uint8_t *pageCount, uint32_t num_items, parser_context_t *ctx) {
+
+    uint32_t start = ctx->offset;
+
+    char buffer[300];
+    MEMZERO(buffer, 300);
```

Comment:
Buffer will be zeroize within for loop. We could initialize char **buffer[300] = {0};**